oauth2-authorization-server/src/main/resources/org/springframework/security/oauth2/server/authorization/client/oauth2_registered_client_h2.ddl

```diff
@@ -0,0 +1,16 @@
+create table oauth2_registered_client (
+    id varchar(128) primary key not null,
+    client_id varchar(128) not null,
+    client_id_issued_at datetime,
+    client_secret varchar(4096) not null,
+    client_secret_expires_at datetime,
+    client_name varchar(512),
+    client_authentication_methods varchar(512) not null,
+    authorization_grant_types varchar(512) not null,
+    redirect_uris varchar(4096) not null,
+    scopes varchar(1024) not null,
+    require_proof_key boolean not null,
+    require_user_consent boolean not null,
+    access_token_ttl integer default 300000 not null,
+    reuse_refresh_tokens boolean default true not null,
+    refresh_token_ttl integer default 600000 not null);
```

Comment:
For the settings, please consider using a JSON field, `client_settings varchar(1000) DEFAULT NULL` and `token_settings varchar(1000) DEFAULT NULL`. See #304 for an example of this (e.g. `attributes`).

Reply:
Implemented but I have the same remarks as with client_secret LOB. It is more complex both in code and in day-to-day maintenance if application administrator wants to make manual changes in database.